openmc/deplete/results_list.py

```diff
@@ -192,3 +193,72 @@ def get_depletion_time(self):
+
+        Parameters
+        ----------
+        burn_index : int
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            nuclides list to evaluate with neutron data. If not provided,
+            nuclides from the cross_sections element of materials.xml will
+            be used. If that element is not present, nuclides from
+            OPENMC_CROSS_SECTIONS will be used.
+        new_xml_name : str, optional
+            name of xml file to put depleted materials into. If not
+            provided, no new xml file will be written.
+
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = openmc.Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            available_cross_sections = nuc_with_data
+        else:
+            # select cross_sections.xml file to use
+            if mat_file.cross_sections:
+                this_library = DataLibrary.from_xml(path=mat_file.cross_sections)
+            else:
+                this_library = DataLibrary.from_xml()
+
+            # Find neutron libraries we have access to
+            available_cross_sections = []
+            for lib in this_library.libraries:
+                if lib['type'] == 'neutron':
+                    available_cross_sections.extend(lib['materials'])
+            if not available_cross_sections:
+                raise Exception('No neutron libraries found in cross_sections.xml')
+
+        # Overwrite material definitions, if they can be found in the depletion
+        # results, and save them to the new depleted xml file.
+        for mat in mat_file:
+            if str(mat.id) in result.mat_to_ind.keys():
+                mat.volume = result.volume[str(mat.id)]
+                for nuc in result.nuc_to_ind.keys():
+                    atoms = result[0, str(mat.id), nuc]
+                    if atoms > 0.0:
+                        atoms_per_barn_cm = 1e-24 * atoms / mat.volume
+                        if nuc in available_cross_sections:
+                            mat.add_nuclide(nuc, atoms_per_barn_cm)
```

Comment:
I'd prefer to place this section earlier in the loop, and then continuing if the nuclide does not have xs. Skip out on some indexing and minor math operations on the assumption that more nuclides will be present in the depletion chain than have cross sections. Might not hold for reduced chains like CASL though

Reply:
Fair enough. I think this is a very reasonable change. I've added it.